core/src/main/scala/shapeless/typeable.scala

```diff
@@ -396,8 +396,14 @@ class TypeableMacros(val c: blackbox.Context) extends SingletonTypeUtils {
 
         if(!pSym.isCaseClass)
           c.abort(c.enclosingPosition, s"No default Typeable for parametrized type $tpe")
-        val fields = tpe.decls.toList collect {
-          case sym: TermSymbol if sym.isVal && sym.isCaseAccessor => sym.typeSignatureIn(tpe)
+        val fields = tpe.decls.sorted collect {
```

Comment:
Good catch! There are probably a couple of other places where `sorted` should be used. Would you mind creating a ticket for that so that it doesn't get forgotten about?